.github/workflows/checkstyle-check.yml

```diff
@@ -0,0 +1,64 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# This workflow will build a Java project with Maven
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-maven
+
+name: Checkstyle Check
+
+on:
+  push:
+    branches: [ master, dev ]
+    paths:
+      - '**/src/main/**'
+      - '**/src/test/**'
+      - '.github/workflows/checkstyle-check.yml'
+      - '!examples/**'
+      - '!*.md'
+  pull_request:
+    branches: [ master ]
+    paths:
+      - '**/src/main/**'
+      - '**/src/test/**'
+      - '.github/workflows/checkstyle-check.yml'
+      - '!examples/**'
+      - '!*.md'
+
+concurrency:
+  group: ${{ github.event_name }}-${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  checkstyle-check:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - uses: actions/checkout@v3
+      - name: Cache Maven Repos
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: shardingsphere-it-cache-${{ github.sha }}
```

Comment:
Why the key is IT?

Reply:
I was thinking about how to caching the plugin related but not the rest dependencies.
this is a good refactor point.